test/shape_test.cpp

```diff
@@ -956,4 +956,24 @@ TEST_CASE(test_multi_index)
     EXPECT(migraphx::verify_range(s.multi(34), std::vector<size_t>{1, 1, 4}));
 }
 
+TEST_CASE(from_2d_permutation)
+{
+    std::vector<std::size_t> out_lens = {2, 3};
+    std::vector<int64_t> permutation  = {1, 0};
+    migraphx::shape out_shape =
+        migraphx::shape::from_permutation(migraphx::shape::float_type, out_lens, permutation);
+    EXPECT(migraphx::verify_range(out_shape.lens(), out_lens));
+    EXPECT(migraphx::verify_range(out_shape.strides(), std::vector<std::size_t>{1, 2}));
```

Comment:
Dont check the strides directly. Instead check `find_permutation(out_shape) == permutation`.

Reply:
I'm going to make additional tests for `find_permutation(s)`. It's not clear to me what it's doing right now.